src/ui-components/SidebarIcon.ts

```diff
@@ -4,7 +4,13 @@ import { StackedPanel } from '@lumino/widgets';
 
 // it could extend Widget, StackedPanel is just temporary as it gives 0-width layout
 export class SidebarIcon extends StackedPanel {
-  constructor(private _options: { label: string; icon: LabIcon; execute: () => boolean | void }) {
+  constructor(
+    private _options: {
+      label: string;
+      icon: LabIcon;
+      execute: () => boolean | void | Promise<any>;
```

Comment:
I see suspect that you did it to remove intertmediate function in:

```
    app.shell.add(
      new SidebarIcon({
        label: 'Files',
        icon: EverywhereIcons.folderSidebar,
-        execute: () => {
-          app.commands.execute(Commands.openFiles);
-        }
+        execute: () => app.commands.execute(Commands.openFiles)
      }),
```

I would suggest to undo this change because the return value of `execute` is used to cancel the switch and on sidebar for icons without a state (JE icon), and this can break it if someone adds async there which would be now permissible. In other words, it was here for a reason (probable should be documented better) and was not broken :)